packages/grid/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -125,16 +107,111 @@ const GridCell = React.forwardRef<HTMLDivElement, GridCellProps>((props, ref) =>
     ...other
   } = props;
 
-  const valueToRender = formattedValue == null ? value : formattedValue;
+  const field = column.field;
   const cellRef = React.useRef<HTMLDivElement>(null);
   const handleRef = useForkRef(ref, cellRef);
   const focusElementRef = React.useRef<FocusElement>(null);
   const apiRef = useGridApiContext();
-
   const rootProps = useGridRootProps();
+
+  const editRowsState = useGridSelector(apiRef, gridEditRowsStateSelector);
+
+  const { classes: rootClasses, getCellClassName } = rootProps;
+
+  const isSelected = useGridSelector(apiRef, () =>
+    apiRef.current.unstable_applyPipeProcessors('isCellSelected', false, {
+      id: rowId,
+      field,
+    }),
+  );
+
+  const cellParams = useGridSelector(
+    apiRef,
+    () => {
+      // This is required because `.getCellParams` tries to get the `state.rows.tree` entry
+      // associated with `rowId`/`fieldId`, but this selector runs after the state has been
+      // updated, while `rowId`/`fieldId` reference an entry in the old state.
+      try {
+        return apiRef.current.getCellParams<any, any, any, GridTreeNodeWithRender>(rowId, field);
+      } catch (e) {
+        if ((e as Error).message.startsWith('No row with id')) {
+          return null as any;
+        }
+        throw e;
+      }
+    },
+    shallowCompare,
+  );
```

Comment:
This is quite ugly. The problem is that `.getCellParams()` reaction to accessing a cell that doesn't exist anymore is to throw an error. For selectors, we'd need a function that returns an empty value (`null` or `undefined`) if accessing a cell that doesn't exist.

The `catch` branch is hit when there is a cell that has been deleted from the state but the cell component hasn't been unmounted yet.

I'm open to suggestions if someone wants this to be prettier.

Reply:
I think it's fine the way it is 👍

Reply:
Yeah I don't want to change it, changing the behavior/signature of `.getCellParams` would be semver major :|

Reply:
Here's some context on why `getCellParams` throws an error if the row is missing https://github.com/mui/mui-x/pull/2010

Reply:
Maybe use a custom error that extends `Error` and use `instanceof` instead of checking the error message?